NN-notebook-v7/notebook-v7.md

```diff
@@ -0,0 +1,118 @@
+
+The following Notebook v6 extensions have been flagged by the community as critical to several major deployments and widely-used workflows. We will work with the extension developers to help port these (or help make new equivalent extensions) for Notebook v7:
+
+- [Nbgrader](https://github.com/jupyter/nbgrader)
+- [RISE](https://github.com/damianavila/RISE)
+- [Jupytext](https://github.com/mwouts/jupytext/issues/271)
+- IPython Parallel (filebrowser tab)
+
+### Extensions whose functionality will be built into Notebook v7
+
+The following extensions will have largely equivalent experiences shipping in core Notebook v7. We will need to track whether there are any remaining differences in their functionality that are either regressions, or potentially disrupting or confusing to existing users.
+
+- [Table of Contents](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/toc2)
+- [Collapsible headings](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/collapsible_headings)
+
+## Major new features in Notebook v7
+
+In addition to having parity with the Notebook v6 user experience and widely-used extensions, Notebook v7 will inherit major new features that have been developed in the JupyterLab project, thus providing automatically a number of improvements to Notebook users as they transition to v7:
+
+- Debugger
+- Real-time collaboration
+- Theming and dark mode
+- Internationalization
+- Improved Accessibility
+- Support for many JupyterLab extensions, including Jupyter LSP (Language Server Protocol) for enhanced code completions
+
+## Technical details
+
+- [RetroLab](https://github.com/jupyterlab/retrolab) running on [Jupyter Server](https://github.com/jupyter-server/jupyter_server) will be the basis for Notebook v7.
+- The current [RetroLab](https://github.com/jupyterlab/retrolab) commit history will be grafted into the [Jupyter Notebook repository](https://github.com/jupyter/notebook) in a pull request.
+- Notebook v7 will continue to be released as the `notebook` Python package.
+- Issues and PRs in the [Notebook repository](https://github.com/jupyter/notebook) will be triaged in this transition.
+
+## Timeline
+
+We anticipate that Notebook v7 will be released in 2022.
+
+Jupyter Notebook v6 will be maintained with critical security fixes for two years following the release of Jupyter Notebook v7.
+
+The [`nbclassic`](https://github.com/jupyterlab/nbclassic) Python package (which provides Jupyter Notebook version 6 on top of Jupyter Server) will also be maintained with critical security fixes for two years following the release of Notebook v7.  However, upon release of Notebook v7, the default URL route for accessing the classic experience will be served by the new code, and nbclassic will likely use a new route. The details of this transition will be worked out in the coming period.
```

Comment:
@Zsailer is this reasonable?

Reply:
Since `nbclassic` can never be installed in the same environment as notebook v7, I don't think we need its URL routes.

Reply:
To clarify, `nbclassic` depends on `notebook <= 6` python package (for its Javascript). 

If `nbclassic` is installed, but `notebook >= 7` is also installed, `nbclassic` will not work. The `notebook` package won't contain the old Notebook JavaScript anymore.

We can, of course, provide the old notebook Javascript from a separate package, but I'm not sure we want to do this. I think nbclassic becomes almost irrelevant once we release notebook v7. It's really just a way to run old notebook UI on the newer jupyter server.

Reply:
`nbclassic` is two things:
1. A server extension that provides the old notebook UI on Jupyter Server. 
2. A shim layer to port old notebook **server** traits to Jupyter Server. 

I think (1) should go away in notebook v7; I feel less strongly about (2).